qualtran/bloqs/phase_estimation/text_book_qpe.py

```diff
@@ -220,13 +222,14 @@ def decompose_from_registers(
         self, context: cirq.DecompositionContext, **quregs
     ) -> cirq.OP_TREE:
         target_quregs = {reg.name: quregs[reg.name] for reg in self.target_registers}
-        unitary_op = self.unitary.on_registers(**target_quregs)
+        target_qubits = merge_qubits(self.target_registers, **target_quregs)
 
-        phase_qubits = quregs['qpe_reg']
+        phase_qubits: Sequence[cirq.Qid] = quregs['qpe_reg'].flatten()
 
         yield self.ctrl_state_prep.on(*phase_qubits)
         for i, qbit in enumerate(phase_qubits[::-1]):
-            yield cirq.pow(unitary_op.controlled_by(qbit), 2**i)
+            yield Power(self.unitary.controlled(), 2**i).on(qbit, *target_qubits)
+            # yield cirq.pow(unitary_op.controlled_by(qbit), 2**i)
```

Comment:
remove old code

Reply:
Reverted the file so now there are no changes and the original implementation uses bloqs instead of cirq gates.